fix: since 5.3.13  Cannot find module '/var/task/node_modules/nuxtseo-shared/dist/utils.mjs'

### 🐛 The bug

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/node_modules/nuxtseo-shared/dist/utils.mjs' imported from /var/task/index.mjs
    at finalizeResolution (node:internal/modules/esm/resolve:281:11)
    at moduleResolve (node:internal/modules/esm/resolve:867:10)
    at moduleResolveWithNodePath (node:internal/modules/esm/resolve:991:14)
    at defaultResolve (node:internal/modules/esm/resolve:1034:79)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:747:20)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:724:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:320:38)
    at ModuleJob._link (node:internal/modules/esm/module_job:182:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:103:5) {
code: "ERR_MODULE_NOT_FOUND",
url: "file:///var/task/node_modules/nuxtseo-shared/dist/utils.mjs"
}
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.

### 🛠️ To reproduce

x

### 🌈 Expected behavior

no errors

### ℹ️ Additional context

_No response_

--- a/.github/workflows/performance.yml
+++ b/.github/workflows/performance.yml
@@ -48,6 +48,7 @@ jobs:
           cp -R "$FIXTURE_SOURCE/." packages/nuxt-seo/test/fixtures/performance/
           pnpm install --frozen-lockfile --trust-lockfile
           pnpm dev:prepare
+          pnpm --dir packages/shared build
           pnpm --filter @nuxtjs/seo exec nuxi build test/fixtures/performance
 
       - name: Warn when the base build is unavailable

--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
   "scripts": {
     "build": "pnpm --filter './packages/*' build",
     "benchmark": "node bench/performance/run.mjs --server packages/nuxt-seo/test/fixtures/performance/.output/server/index.mjs --output bench/performance/result.json --profiles bench/performance/profiles",
-    "benchmark:build": "pnpm --filter @nuxtjs/seo exec nuxi build test/fixtures/performance",
+    "benchmark:build": "pnpm --dir packages/shared build && pnpm --filter @nuxtjs/seo exec nuxi build test/fixtures/performance",
     "benchmark:test": "node --test bench/performance/*.test.mjs",
     "dev": "pnpm --filter @nuxtjs/seo dev",
     "dev:prepare": "pnpm --filter './packages/**' stub && pnpm --filter @nuxtjs/seo dev:prepare",

--- a/packages/shared/src/nitro-compatibility.ts
+++ b/packages/shared/src/nitro-compatibility.ts
@@ -35,6 +35,9 @@ const runtimeSetupMarker = Symbol.for('nuxtseo:nitro-runtime-compatibility:reque
 
 interface NuxtNitroCompatibilityOptions {
   alias?: Record<string, string>
+  externals?: {
+    inline?: (string | RegExp)[]
+  }
   typescript?: {
     tsConfig?: {
       compilerOptions?: {
@@ -196,7 +199,12 @@ function applyNitroRuntimeCompatibility(
   const nuxtOptions = nuxt.options as Nuxt['options'] & { nitro?: NuxtNitroCompatibilityOptions }
   const nitroOptions = nuxtOptions.nitro ||= {}
   nitroOptions.alias ||= {}
+  nitroOptions.externals ||= {}
+  nitroOptions.externals.inline ||= []
   nitroOptions.virtual ||= {}
+  // Vercel can omit traced shared subpaths from deployed functions: https://github.com/harlan-zw/nuxt-seo/issues/623
+  if (!nitroOptions.externals.inline.includes('nuxtseo-shared'))
+    nitroOptions.externals.inline.push('nuxtseo-shared')
   const h3RuntimeModule = compatibility._tag === 'nitro-v3' ? 'nitro/h3' : 'h3'
   nitroOptions.alias[H3_RUNTIME_MODULE] = h3RuntimeModule
   const h3Resolution = resolveRuntimeModule(nuxt, h3RuntimeModule)

--- a/packages/shared/test/unit/nitro.test.ts
+++ b/packages/shared/test/unit/nitro.test.ts
@@ -150,7 +150,7 @@ describe('setupNitroRuntimeCompatibility', () => {
     )
   })
 
-  it('registers shared templates once when several modules call setup', () => {
+  it('registers shared compatibility once when several modules call setup', () => {
     getNuxtVersionMock.mockReturnValue('4.5.1')
     const nuxt = createNuxt()
 
@@ -158,6 +158,7 @@ describe('setupNitroRuntimeCompatibility', () => {
     setupNitroRuntimeCompatibility(nuxt)
 
     expect(addTypeTemplateMock).toHaveBeenCalledOnce()
+    expect(nuxt.options.nitro.externals?.inline).toEqual(['nuxtseo-shared'])
   })
 
   it('registers the request context type bridge after an older helper ran', async () => {

